File: ScoutTrack/ScoutTrack.Services/Database/Entities/Troop.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ScoutTrack.Services.Database.Entities
{
    public class Troop : UserAccount
    {
        [Required]
        [MaxLength(100)]
        public string Name { get; set; } = string.Empty;

        [ForeignKey(nameof(City))]
        public int CityId { get; set; }
        public City? City { get; set; }

        [Required]
        public double Latitude { get; set; }

        [Required]
        public double Longitude { get; set; }

        [Phone]
        [MaxLength(20)]
        public string ContactPhone { get; set; } = string.Empty;

        [MaxLength(100)]
        public string ScoutMaster { get; set; } = string.Empty;  // Starješina

        [MaxLength(100)]
        public string TroopLeader { get; set; } = string.Empty;  // Načelnik

        public DateTime FoundingDate { get; set; } = DateTime.UtcNow;

        public string LogoUrl { get; set; } = string.Empty;
        public ICollection<Member> Members { get; set; } = new List<Member>();
        public ICollection<Activity> Activities { get; set; } = new List<Activity>();
    }
}

File: ScoutTrack/ScoutTrack.UI/scouttrack_desktop/lib/models/troop.dart
import 'package:json_annotation/json_annotation.dart';

part 'troop.g.dart';

@JsonSerializable()
class Troop {
  final int id;
  final String username;
  final String email;
  final String name;
  final int cityId;
  final String cityName;
  final double latitude;
  final double longitude;
  final String contactPhone;
  final String logoUrl;
  final bool isActive;
  final DateTime createdAt;
  final DateTime? updatedAt;
  final DateTime? lastLoginAt;
  final int memberCount;
  final String scoutMaster;
  final String troopLeader;
  final DateTime foundingDate;

  Troop({
    this.id = 0,
    this.username = '',
    this.email = '',
    this.name = '',
    this.cityId = 0,
    this.cityName = '',
    this.latitude = 0,
    this.longitude = 0,
    this.contactPhone = '',
    this.logoUrl = '',
    this.isActive = false,
    required this.createdAt,
    this.updatedAt,
    this.lastLoginAt,
    this.memberCount = 0,
    this.scoutMaster = '',
    this.troopLeader = '',
    required this.foundingDate,
  });

  factory Troop.fromJson(Map<String, dynamic> json) {
    return _$TroopFromJson(json);
  }
}